optimum/habana/transformers/models/blip/modeling_blip_text.py

```diff
@@ -258,7 +258,8 @@ def gaudi_BlipTextEncoder_forward(
             next_decoder_cache += (layer_outputs[-1],)
         if output_attentions:
             all_self_attentions = all_self_attentions + (layer_outputs[1],)
-            all_cross_attentions = all_cross_attentions + (layer_outputs[2],)
+            if all_cross_attentions is not None:
```

Comment:
This change looks OK, although in next transformers upgrade we will have already this code protected by different conditional but the effect will be the same.

https://github.com/huggingface/transformers/commit/c8524aeb07f370195733f610760236d22e88cb61#diff-00846f08e1b2a41509f5f669a49fc36baac8555a83da24c30f8ac9e7a9024d59R472